project_euler/problem_074/sol2.py

```diff
@@ -36,18 +40,22 @@ def factorial(a: int) -> int:
     if a < 0:
         raise ValueError("Invalid negative input!", a)
 
-    # The case of 0! is handled separately
-    if a == 0:
-        return 1
-    else:
-        # use a temporary support variable to store the computation
-        temporary_computation = 1
+    if not FACTORIAL_CACHE.get(a):
+        # The case of 0! is handled separately
+        if a == 0:
+            FACTORIAL_CACHE[a] = 1
```

Comment:
Please change this back to `return 1`

Reply:
Not yet resolved.

Reply:
uhhmm, any reason why we don't want to save 0 is the cache dict?

Reply:
I thought of something else. No need to change this one.